### PR TITLE
fix(ecs-patterns): Fix Network Load Balancer Port assignments in ECS Patterns

### LINKED ISSUE #18073: (ecs-patterns): NetworkLoadBalancedServiceBase and NetworkMultipleTargetGroupsServiceBase hardcode port 80 on target group
### What is the problem?

As stated in the title, both base classes default to port 80 for the Loadbalancer TargetGroup.
The culprit for NetworkLoadBalancedServiceBase is located here: https://github.com/aws/aws-cdk/blob/v2-main/packages/@aws-cdk/aws-ecs-patterns/lib/base/network-load-balanced-service-base.ts#L345

it needs to be changed to something like this:
```typescript
const targetProps = {
      port: props.taskImageOptions?.containerPort ?? 80,
    };
```

The culprit for NetworkMultipleTargetGroupsServiceBase is located here: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts#L377

it needs to be
```typescript
        port: targetProps.containerPort ?? 80,
```

to result in

```json
{
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
      "Properties": {
        "Port": [Custom Port],
        "Protocol": "TCP",
      },
    }
```

### Reproduction Steps

I will provide a sample for the NetworkLoadBalancedServiceBase bug:

```typescript
import { Stack, StackProps } from 'aws-cdk-lib';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Cluster, ContainerImage } from 'aws-cdk-lib/aws-ecs';
import { NetworkLoadBalancedFargateService } from 'aws-cdk-lib/aws-ecs-patterns';
import { Construct } from 'constructs';

export class ElbTestStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    const vpc = new Vpc(this, 'vpc', {});

    const fargateCluster = new Cluster(this, 'FarGate-Cluster', {
      clusterName: 'test',
      vpc: vpc,
      enableFargateCapacityProviders: true,
      containerInsights: true
    })

    const loadBalancedFargateService = new NetworkLoadBalancedFargateService(this, 'NLBService', {
      cluster: fargateCluster,
      memoryLimitMiB: 1024,
      cpu: 512,
      taskImageOptions: {
        image: ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
        containerPort: 81
      },
      listenerPort: 8181,
    });

  }
}
```

### What did you expect to happen?

I expected the Target Group to look like this:
```json
"NLBServiceLBPublicListenerECSGroup1257E89B": {
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
      "Properties": {
        "Port": 81,
(snip)
      },
      "Metadata": {
        "aws:cdk:path": "ElbTestStack/NLBService/LB/PublicListener/ECSGroup/Resource"
      }
    }
```

### What actually happened?

The Target Group looked like this:

```json
"NLBServiceLBPublicListenerECSGroup1257E89B": {
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
      "Properties": {
        "Port": 80,
(snip)
      },
      "Metadata": {
        "aws:cdk:path": "ElbTestStack/NLBService/LB/PublicListener/ECSGroup/Resource"
      }
    }
```

when implementing 
```typescript
const targetProps = {
      port: props.taskImageOptions?.containerPort ?? 80,
    };
```
the bug was fixed.

### CDK CLI Version

2.2.0 (build 4f5c27c)

### Framework Version

_No response_

### Node.js Version

v16.13.0

### OS

WSL 2 on Ubuntu

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by LukvonStrom**
This PR introduces changeable ports as a regression fix for the hardcoded port 80 in both NLB constructs bases.

Closes #18073 

Additionally it seems like the regression reported in the linked issue was not spotted in the integration tests either. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by LukvonStrom** [comment]
hi @madeline-k, were you able to look over this? :) I need this PR in prod for a use case so any feedback towards a merge would be highly appreciated!

**Comment by LukvonStrom** [comment]
cc/ @skinny85 as we had the pleasure already in the other PR :)

**Comment by skinny85** [comment]
This is not my area of expertise @LukvonStrom, but I'll try to get the right people involved 🙂,

File: packages/@aws-cdk/aws-ecs-patterns/test/fargate/load-balanced-fargate-service-v2.test.ts [hunk 1]
```diff
@@ -661,4 +662,75 @@ describe('When Network Load Balancer', () => {
       });
     }).toThrow(/You must specify one of: taskDefinition or image/);
   });
+
+  test('test Fargate networkloadbalanced construct with custom Port', () => {
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs-patterns/test/fargate/load-balanced-fargate-service-v2.test.ts:666]: After merging the latest from master, you will need to use the Assertions unit test framework for the new tests. I migrated the unit tests in this module earlier this week: https://github.com/aws/aws-cdk/pull/18520/files. The changes you will need to make should be super minimal.

**Review by madeline-k — COMMENTED** [review]
Thanks for creating this fix @LukvonStrom! This looks good to me. 

I have one question: Why do you say this is a regression? I am curious when it was introduced.

**Comment by LukvonStrom** [comment]
> Thanks for creating this fix @LukvonStrom! This looks good to me.
> 
> I have one question: Why do you say this is a regression? I am curious when it was introduced.

@madeline-k the problem was (re-)introduced in #4988, the initial fix brought forward in #4793 was then patched out again.

File: packages/@aws-cdk/aws-ecs-patterns/test/fargate/load-balanced-fargate-service-v2.test.ts (hunk 1 above)
Comment (LukvonStrom) [reply packages/@aws-cdk/aws-ecs-patterns/test/fargate/load-balanced-fargate-service-v2.test.ts:666]: Done! :)

**Comment by madeline-k** [comment]
> > Thanks for creating this fix @LukvonStrom! This looks good to me.
> > I have one question: Why do you say this is a regression? I am curious when it was introduced.
> 
> @madeline-k the problem was (re-)introduced in #4988, the initial fix brought forward in #4793 was then patched out again.

Ah, thanks for the context! Looks like in https://github.com/aws/aws-cdk/issues/4983 (fixed by #4988), the issue was that the wrong Protocol was being used. And this PR only fixes things so that the correct Port is used. So, we won't be re-introducing issue #4983 here.

**Review by madeline-k — APPROVED** [review]
Thanks @LukvonStrom!

**Pull request merged**